src/librustc_mir/borrow_check/nll/region_infer/values.rs

```diff
@@ -50,90 +51,65 @@ impl RegionValueElements {
+    fn point_from_location(&self, location: Location) -> PointIndex {
+        let Location {
+            block,
+            statement_index,
+        } = location;
+        let start_index = self.statements_before_block[block];
+        PointIndex::new(start_index + statement_index)
     }
 
-    /// Iterates over the `RegionElementIndex` for all points in the CFG.
-    crate fn all_point_indices<'a>(&'a self) -> impl Iterator<Item = RegionElementIndex> + 'a {
-        (0..self.num_points).map(move |i| RegionElementIndex::new(i + self.num_universal_regions))
+    /// Range coverting all point indices.
+    fn all_points(&self) -> Range<PointIndex> {
+        PointIndex::new(0)..PointIndex::new(self.num_points)
     }
 
-    /// Converts a particular `RegionElementIndex` to the `RegionElement` it represents.
-    crate fn to_element(&self, i: RegionElementIndex) -> RegionElement {
-        debug!("to_element(i={:?})", i);
-
-        if let Some(r) = self.to_universal_region(i) {
-            RegionElement::UniversalRegion(r)
-        } else {
-            let point_index = i.index() - self.num_universal_regions;
-
-            // Find the basic block. We have a vector with the
-            // starting index of the statement in each block. Imagine
-            // we have statement #22, and we have a vector like:
-            //
-            // [0, 10, 20]
-            //
-            // In that case, this represents point_index 2 of
-            // basic block BB2. We know this because BB0 accounts for
-            // 0..10, BB1 accounts for 11..20, and BB2 accounts for
-            // 20...
-            //
-            // To compute this, we could do a binary search, but
-            // because I am lazy we instead iterate through to find
-            // the last point where the "first index" (0, 10, or 20)
-            // was less than the statement index (22). In our case, this will
-            // be (BB2, 20).
-            //
-            // Nit: we could do a binary search here but I'm too lazy.
-            let (block, &first_index) = self
-                .statements_before_block
-                .iter_enumerated()
-                .filter(|(_, first_index)| **first_index <= point_index)
-                .last()
-                .unwrap();
-
-            RegionElement::Location(Location {
-                block,
-                statement_index: point_index - first_index,
-            })
-        }
-    }
-
-    /// Converts a particular `RegionElementIndex` to a universal
-    /// region, if that is what it represents. Returns `None`
-    /// otherwise.
-    crate fn to_universal_region(&self, i: RegionElementIndex) -> Option<RegionVid> {
-        if i.index() < self.num_universal_regions {
-            Some(RegionVid::new(i.index()))
-        } else {
-            None
+    /// Converts a particular `RegionElementIndex` to a location, if
+    /// that is what it represents. Returns `None` otherwise.
```

Comment:
The sentence "Returns `None` otherwise." Can't be right since you've made this return `Location`, not an `Option<_>`...

Reply:
Yeah, outdated. Will fix.